metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/resolver.py

```diff
@@ -1076,6 +1080,27 @@ def create_urn_using_old_parser(self, query: str, server: str) -> Lineage:
             column_lineage=[],
         )
 
+    def get_db_name(self, data_access_tokens: List[str]) -> Optional[str]:
+        # In-case of DatabricksMultiCloud_SQL the database-name is the catalog name,
+        # and that name is not available in SQL statement
+        if (
+            data_access_tokens[0]
+            != SupportedDataPlatform.DatabricksMultiCloud_SQL.value.powerbi_data_platform_name
+        ):
+            return None
+
+        if (
+            len(data_access_tokens) >= 13
+        ):  # Explicit catalog name is set in Database argument
+            return tree_function.strip_char(data_access_tokens[9])
+
+        if (
+            len(data_access_tokens) >= 6 and data_access_tokens[4] == "Catalog"
+        ):  # use Catalog name is database
+            return tree_function.strip_char(data_access_tokens[5])
```

Comment:
This looks super magical, could you add some example lists of tokens to explain what cases are being handled?